### PR TITLE
NEW: LP-19. Implement Permission logic to enable/disable CM Landing Page on the ugly Login Page

### PR DESCRIPTION
**Pull request by denys-davydov**

### CODE REVIEW

File: spec/services/content/content_service_spec.rb [hunk 1]
```diff
@@ -24,8 +24,78 @@ module Content
...
+
+        it 'doesn\'t filter Case Management service from content when profile has \'CWS Case Management System\' privilege' do
+          profile = { :roles => [], :privileges => ['CWS Case Management System'] }
+          actual_result = test_subject.filter_content(profile)
+          filtered_result_size = actual_result[:services].select{|e| e['id'] == 'case_management'}.size
+          expect(filtered_result_size).to eq 1
+        end
+      end
+
+      describe '#is_permitted?' do
```
Comment (edgibbs) [inline spec/services/content/content_service_spec.rb]: This is a private method, should test the public api `filter_content()'

`send` is fun though.

File: app/services/content/content_service.rb [hunk 2]
```diff
@@ -14,5 +21,29 @@ def content
...
+          .select { |service| is_permitted?(service, profile) },
+        :resources => content[:resources]
+          .select { |resource| is_permitted?(resource, profile) },
+      }
+    end
+
+    def is_permitted?(service, profile)
+      roles = service['roles']
+      privileges = service['privileges']
+      return true if is_empty?(roles) && is_empty?(privileges)
```
Comment (edgibbs) [inline app/services/content/content_service.rb]: ActiveSupport, one of the gems you get with rails has a `#blank?` method which checks strings and arrays for if they're nil or empty, so you can just refactor as:

```ruby
  return true if roles.blank? && privileges.blank?
```

File: spec/services/content/content_service_spec.rb (hunk 1 above)
Comment (denys-davydov) [reply spec/services/content/content_service_spec.rb]: Well, I have 2 basic tests for the public `#filter_content()` which show that content is filtered for some cases and not filtered for different ones,  and 8 tests for the private method `#is_permitted?()` which cover all the cases of filtering with all the corner cases.

File: app/services/content/content_service.rb (hunk 2 above)
Comment (denys-davydov) [reply app/services/content/content_service.rb]: Updated the line

File: spec/services/content/content_service_spec.rb (hunk 1 above)
Comment (denys-davydov) [reply spec/services/content/content_service_spec.rb]: But it will become cumbersome and harder to understand and support the test code. Now it is easy, straightforward and covers all the cases. 
I understand that testing private methods is considered to be a bad practice or anti-pattern, but I would prefer to have a clean, easy to read and support code as a trade-off for breaking some good practice rules. Of course, I can create a new class, move the method there, make the method public and test it in "good practice" paradigm, but do I really have to?
